File: frontend/src/objects/rider/profile/RiderPerformance.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type {
  CurrencyCents,
  EntityCount,
  RatingValue,
} from '@/objects/core/SharedObjects'

export type RiderPerformance = {
  averageRating: RatingValue
  ratingCount: EntityCount
  oneStarRatingCount: EntityCount
  earningsCents: CurrencyCents
}

File: backend/src/main/scala/services/rider/objects/RiderPerformance.scala
package services.rider.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given

import system.objects.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*

final case class RiderPerformance(
    averageRating: AverageRating,
    ratingCount: EntityCount,
    oneStarRatingCount: EntityCount,
    earningsCents: CurrencyCents,
)
object RiderPerformance:
  given Encoder[RiderPerformance] = deriveEncoder
  given Decoder[RiderPerformance] = deriveDecoder
